### ISSUE TITLE
Joystick prompt not useable for RZX

### ISSUE THREAD

**Issue by ghost** [bog-dan-ro/spectacol#87]
Sorry I opened another issue for this because I don't think the Joystick prompt should come up or alter your set joystick setting when watching RZX. If the RZX used a Cursor joystick and you change it to Kempston in the Joystick prompt, when you stop the playback the joystick doesn't work. You can only use the Joystick type that they selected in the RZX which makes the joystick prompt useless.

If you have the Joystick prompt turned off and load a RZX it always changes your set joystick to Cursor even if they used Sinclair 1 in the RZX.

So to sum up, I think it should not change what your joystick is set at when watching RZX unless you manually change it yourself.

**Comment by bog-dan-ro** [comment]
You're right, I'll upstream the problem (because the joystick prompt comes from FUSE).

**Comment by bog-dan-ro** [comment]
Upstreamed to https://sourceforge.net/p/fuse-emulator/bugs/355/

**Comment by ghost** [comment]
The joystick is also changed when loading z80 etc. with the 'Snapshot Joystick Prompt' OFF. I don't think this should happen. This doesn't happen on FUSE.

**Comment by ghost** [comment]
This is how I understand it in layman's terms and what should be happening in Spectacol.

If the joystick prompt is off :-
1) The user's joystick is set to 'Kempston' for example.
2) They load a snapshot that was made with 'Cursor' joystick for example.
3) Nothing changes and the user's joystick stays as 'Kempston'.

If the joystick prompt is turned on :-
1) The user's joystick is set to 'Kempston' for example.
2) They load a snapshot that was made with 'Cursor' joystick for example.
3) Because the user's joystick is different a joystick prompt appears telling the user that 'Cursor' joystick was used and do they want to change to it. No other joystick choices, just a yes or no.

This should happen for all snapshots.

**Comment by ghost** [comment]
How's everything going ? It's been very quiet here lately.

This joystick prompt thing is wrong in Spectacol and doesn't happen in Fuse. Can it be changed to the way I described in the post above this one ?

**Comment by ghost** [comment]
Everything OK Bogdan ?

**Comment by bog-dan-ro** [comment]
Yep, everything is ok, I'm just very busy at work :). I fixed this patch in spectacol (when you load a RZX it will change the joystick type silently).
Hopefully today I'll push a new beta version with this fix and a few more from FUSE (e.g. https://sourceforge.net/p/fuse-emulator/bugs/356/ and  https://sourceforge.net/p/fuse-emulator/feature-requests/90/ ).

**Comment by ghost** [comment]
When you load a RZX it will change the joystick type silently ? I think it would be better like this : 

If the joystick prompt is off :-
1) The user's joystick is set to 'Kempston' for example.
2) They load a snapshot that was made with 'Cursor' joystick for example.
3) Nothing changes and the user's joystick stays as 'Kempston'.

If the joystick prompt is turned on :-
1) The user's joystick is set to 'Kempston' for example.
2) They load a snapshot that was made with 'Cursor' joystick for example.
3) Because the user's joystick is different a joystick prompt appears telling the user that 'Cursor' joystick was used and do they want to change to it. No other joystick choices, just a yes or no.

This should happen for all snapshots. This is how it works on Fuse.

**Comment by bog-dan-ro** [comment]
I think your examples are not ok ...
If you saved a RZX snapshot in the middle of a game and you selected **Cursor**, when you load it again if you have **Kempston** selected and I'll keep **Kempston** you'll not be able to play the game when you'll resume it, because the RZX snapshot expects to have a **Cursor** joystick not **Kempston**. Does it make sense ? :)

So, for RZX **only**, when you load it, I'll use the joystick from the snapshot. If prompt is turned on, I'll switch to **Cursor** silently.

**Comment by ghost** [comment]
OK, so it will only change silently when the joystick prompt is set ON ? What about other snapshots like normal Z80 etc. ? will it use the method I posted above ?

**Comment by bog-dan-ro** [comment]
On non RZX snapshots will be as it is now. If the joystick prompt is set ON, it will change to the one from the snapshot and it will show the joystick selection dialog.

**Comment by ghost** [comment]
But it changes the joystick even if the Joystick Prompt if OFF. Bug ?

Why don't you do it the same as Fuse ? There isn't much point in giving a choice of joysticks when it will only use the joystick it was saved with (unless it's changed in-game). If they saved it with Sinclair 1 then only Sinclair 1 will work until it's changed in-game.

**Comment by bog-dan-ro** [comment]
Is not a bug, if the joystick prompt is off, I just use the joystick from the snapshot.
There are two cases here:
- the snapshot is saved at very beginning when the game is not started yet and no joystick is selected.In this case changing the users selected joystick with the one from snapshot will confuse the user. IMHO the snapshot is wrong in this case, because it should not contain any joystick selection.
- the snapshot is saved **after** the game is started with some joystick selected and if I'll keep the users selected one, the game will not work at all, that's why if the joystick prompt is off I must use the joystick selection from snapshot.

**Comment by ghost** [comment]
Why is this completely different to Fuse (and confusing) ? If the joystick prompt is OFF then it should never change your selected joystick, I thought that was the whole point. If the Joystick prompt is ON and you load a Z80 that was saved with a different joystick than your currently selected one then it would Prompt and tell you the joystick that was used is different and do you want to change to it. Have you tried this in Fuse ? It works well. At the moment, Spectacol keeps changing the joystick whether the prompt is on or off. This is really confusing and not needed. There's no need to prompt and give you a choice of all the joysticks when you can change it in the Joystick Type menu anyway. Like you have stated above, the prompt is only useful if the snapshot was saved after the control choice. If you resume recording an RZX and your joystick is different then force change it.
